contracts/interfaces/IEntryPoint.sol

```diff
@@ -189,4 +189,14 @@ interface IEntryPoint is IStakeManager, INonceManager {
      */
     function getSenderAddress(bytes memory initCode) external;
 
+    error DelegateAndRevert(bool success, bytes ret);
+
+    /**
+     * Helper method for dry-run testing.
+     * @dev calling this method, the EntryPoint will make a delegatecall to the given data, and report (via revert) the result.
+     *  The method always revert, so is only useful off-chain for dry run calls, in cases where state-override to replace
+     *  actual EntryPoint code is less convenient.
+     */
+    function delegateAndRevert(bytes calldata data) external;
```

Comment:
Is there a reason to have this method as part of the `EntryPoint` interface? I think this is just a temporary workaround and should not affect the interface at all.

Reply:
yes, we removed all opinionated "view" functions from the EntryPoint, since "state-override" allows using custom functionalities.
But in case state override doesn't exist, this is the minimum required code in entrypoint to allow running off-chain code from entrypoint address without causing any side-effect.

e.g. state-override is not supported by estimateGas, at least on some node implementations. Bundlers can overcome this by rapid RPC "eth_call" calls to the node which they are close to - but end-users can't.